Examples/VisionExamples/Examples/POIDrawingViewController.swift

```diff
@@ -0,0 +1,152 @@
+                                                true).first!
+        let path = documentsPath.appending("/poi-drawing")
+
+        // create VisionReplayManager with a path to recorded session
+        visionManager = try? VisionReplayManager.create(recordPath: path)
+        // register its delegate
+        visionManager.delegate = self
+
+        // configure Vision view to display sample buffers from video source
+        visionViewController.set(visionManager: visionManager)
+        // add Vision view as a child view
+        addVisionView()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        visionManager.start()
+    }
+
+    override func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+        visionManager.stop()
+    }
+
+    private func addVisionView() {
+        addChild(visionViewController)
+        view.addSubview(visionViewController.view)
+        visionViewController.didMove(toParent: self)
+    }
+
+    deinit {
+        // free up VisionManager's resources
+        visionManager.destroy()
+    }
+
+    private func updatePOI(geoCoordinate: GeoCoordinate, poiView: UIView) {
+        // hide the view if one of conditions isn't met
+        let hideView = {
+            poiView.removeFromSuperview()
+        }
+
+        guard
+            // make sure that `Camera` is calibrated for more precise transformations
+            let camera = camera, camera.isCalibrated,
+            // convert geo to world
+            let poiWorldCoordinate = visionManager.geoToWorld(geoCoordinate: geoCoordinate),
+            // make sure POI is in front of the camera and not too far away
+            poiWorldCoordinate.x > 0, poiWorldCoordinate.x < distanceVisibilityThreshold
+        else {
+            hideView()
+            return
+        }
+
+        // by default height of the translated geo coordinate is 0.
+        // If you'd like to lift it above the ground alter its `z` coordinate
+        let worldCoordinateLeftTop =
+            WorldCoordinate(x: poiWorldCoordinate.x,
+                            y: poiWorldCoordinate.y - poiDimension / 2,
+                            z: distanceAboveGround + poiDimension / 2)
+
+        let worldCoordinateRightBottom =
+            WorldCoordinate(x: poiWorldCoordinate.x,
+                            y: poiWorldCoordinate.y + poiDimension / 2,
+                            z: distanceAboveGround - poiDimension / 2)
+
+        guard
+            // convert the POI to the screen coordinates
+            let screenCoordinateLeftTop =
+                visionManager.worldToPixel(worldCoordinate: worldCoordinateLeftTop),
+
+            let screenCoordinateRightBottom =
+                visionManager.worldToPixel(worldCoordinate: worldCoordinateRightBottom)
+        else {
+            hideView()
+            return
+        }
+
+        let frameSize = camera.frameSize.cgSize
```

Comment:
nit: add a small comment (`// drawing the POI on the screen`) to separate blocks logically and describe what's going on next?

Reply:
Done.